core/src/main/java/org/kohsuke/stapler/RequestImpl.java

```diff
@@ -824,8 +825,13 @@ private Object bindResolve(Object o, JSONObject src) {
                         try {
                             Field f = c.getDeclaredField(key);
                             if (f.getAnnotation(DataBoundSetter.class)!=null) {
-                                f.setAccessible(true);
-                                f.set(r, bindJSON(f.getGenericType(), f.getType(), j.get(key)));
+                                try {
+                                    f.set(r, bindJSON(f.getGenericType(), f.getType(), j.get(key)));
+                                } catch(IllegalAccessException e) {
+                                    LOGGER.warning(IllegalReflectiveAccessLogHandler.get(e));
+                                    f.setAccessible(true);
+                                    f.set(r, bindJSON(f.getGenericType(), f.getType(), j.get(key)));
+                                }
```

Comment:
Could move this idiom into a helper method in `IllegalReflectiveAccessLogHandler`.